jsonrpc/types_test.go

```diff
@@ -137,3 +138,63 @@ func TestBlock_Copy(t *testing.T) {
 	bb := b.Copy()
 	require.Equal(t, b, bb)
 }
+
+//go:embed testsuite/*
+var testsuite embed.FS
+
+func TestBlock_Encoding(t *testing.T) {
+	b := block{
+		ParentHash:   types.Hash{0x1},
+		Sha3Uncles:   types.Hash{0x2},
+		Miner:        types.Address{0x1}.Bytes(),
+		StateRoot:    types.Hash{0x4},
+		TxRoot:       types.Hash{0x5},
+		ReceiptsRoot: types.Hash{0x6},
+		LogsBloom:    types.Bloom{0x0},
+		Difficulty:   10,
+		Number:       11,
+		GasLimit:     12,
+		GasUsed:      13,
+		Timestamp:    14,
+		ExtraData:    []byte{97, 98, 99, 100, 101, 102},
+		MixHash:      types.Hash{0x7},
+		Nonce:        types.Nonce{10},
+		Hash:         types.Hash{0x8},
+	}
+
+	testBlock := func(name string) {
+		res, err := json.Marshal(b)
+		require.NoError(t, err)
+
+		data, err := testsuite.ReadFile(name)
+		require.NoError(t, err)
+
+		data = removeWhiteSpace(data)
+		require.Equal(t, res, data)
+	}
+
+	t.Run("empty block", func(t *testing.T) {
+		testBlock("testsuite/block-empty.json")
+	})
+
+	t.Run("block with transactions", func(t *testing.T) {
+		b.Transactions = []transactionOrHash{
+			transactionHash{0x8},
+		}
+		testBlock("testsuite/block-with-txn-hashes.json")
+	})
+
+	t.Run("block with transactions", func(t *testing.T) {
+		// TODO: do the same tests for the transaction object and enable this test
+		t.Skip("TODO")
+	})
```

Comment:
Are we going to implement this test now, otherwise maybe delete it?

You wanted here to create transaction and make sure that nonce is correctly marshaled and unmarshaled (since `Transcation` also has `Nonce`)? In that case, this test doesn't belong here, as this tests block struct.

Reply:
The block can also return the body of the transactions, so we need to make sure that the transactions there are correct too. See comment to Lazar question too.

Reply:
Saw response to Lazar's same question (we were posting comments at about same time :slightly_smiling_face:). 

Although not sure why we don't implement such test right now?

Reply:
This is a bug fix for the block encoding. I will do a followup PR similar to this for transaction.